frameworks/kafka/tests/test_sanity.py

```diff
@@ -341,6 +347,14 @@ def test_metrics():
 
 # --------- Suppressed -------------
 
+@retrying.retry(
+    wait_fixed=10000,
+    stop_max_delay=120000)
+def wait_for_response(suppressed_url):
+    response = dcos.http.get(suppressed_url)
+    response.raise_for_status()
+    assert response.text == "true"
```

Comment:
There is no return statement. Does this not imply infinite retry?

Reply:
it stops retrying once the assertion is not raised. See other comments on explaining this API vs retrying.retry